llvm/lib/CodeGen/WindowScheduler.cpp

```diff
@@ -644,6 +645,20 @@ void WindowScheduler::expand() {
   MSE.cleanup();
 }
 
+void WindowScheduler::reanalyzeLiveIntervals() {
+  auto SlotIndex = Context->LIS->getSlotIndexes();
+  // Check if the SlotIndex infomation is missing.
+  for (auto &MBB : *MF)
+    for (auto &MI : MBB)
+      if (!SlotIndex->hasIndex(MI)) {
+        // The slot index and live intervals of MF have been corrupted and need
+        // to be reanalyzed.
+        SlotIndex->reanalyze(*MF);
+        Context->LIS->reanalyze(*MF);
```

Comment:
This is much too heavy of a hammer. You should be updating specific intervals as you change them (probably need to add some handleMove calls somewhere)

Reply:
You are right. 
The issue occurs in the ModuloScheduleExpander module. We have avoided the current problem by adding the minimal amount of live interval information, without affecting the window scheduler.